File: Nov2025/src/main/java/com/nykaa/pages/LoginPage.java
package com.nykaa.pages;

import org.openqa.selenium.*;
import java.io.IOException;
import com.nykaa.utilities.ActionHelper;

public class LoginPage {
    WebDriver driver;
    ActionHelper helper;

    By signInButton = By.xpath("//button[contains(., 'Sign in') or contains(., 'Login')]");
    By mobileEmailOption = By.xpath("//button[contains(., 'Sign in with Mobile / Email')]");
    By mobileField = By.xpath("//input[@type='tel' or @placeholder='Enter Phone Number']");
    
    By getOtpButton = By.className("css-15q5a8e");
    //By getOtpButton = By.xpath("//button[contains(., 'Send OTP') or contains(., 'Get OTP')]");

    public LoginPage(WebDriver driver) {
        this.driver = driver;
        this.helper = new ActionHelper(driver);
    }

    public void openNykaa() {
        driver.get("https://www.nykaa.com/");
        System.out.println("Opened Nykaa homepage");
    }

    public void loginWithMobile(String phoneNumber) throws IOException, InterruptedException {
        helper.click(signInButton, "ClickSignIn");
        helper.click(mobileEmailOption, "SelectMobileLogin");
        helper.type(mobileField, phoneNumber, "EnterMobileNumber");
        helper.click(getOtpButton, "ClickGetOtp");
        System.out.println("OTP sent — please enter it manually within 30 seconds...");
        Thread.sleep(30000);
    }
}

File: Capstone_Python(Pytest)/pages/login_page.py
import time
from selenium.webdriver.common.by import By


class LoginPage:
    def __init__(self, driver, helper):
        self.driver = driver
        self.helper = helper
        self.sign_in_button = (By.XPATH, "//button[contains(., 'Sign in') or contains(., 'Login')]")
        self.mobile_email_option = (By.XPATH, "//button[contains(., 'Sign in with Mobile / Email')]")
        self.mobile_field = (By.XPATH, "//input[@type='tel' or @placeholder='Enter Phone Number']")
        self.get_otp_button = (By.CLASS_NAME, "css-15q5a8e")

    def open_homepage(self):
        self.driver.get("https://www.nykaa.com/")
        print("🌐 Opened Nykaa homepage")
        self.helper.take_screenshot("OpenNykaa")

    def login(self):
        self.helper.click(self.sign_in_button, "Click Sign In")
        self.helper.click(self.mobile_email_option, "Select Mobile/Email Option")
        self.helper.type(self.mobile_field, "8925334178", "Enter Mobile Number")
        self.helper.click(self.get_otp_button, "Click Get OTP")
        print("⏳ Waiting 30 seconds for manual OTP entry...")
        time.sleep(30)
        self.helper.take_screenshot("LoginSuccess")